test/deployments.ts

```diff
@@ -0,0 +1,135 @@
+import { expect } from "chai";
+import { AbiCoder } from "ethers/lib/utils";
+import { deployments, ethers } from "hardhat";
+import {
+  AludelFactory,
+  CrucibleFactory,
+  MockERC20,
+  PowerSwitchFactory,
+  RewardPoolFactory,
+} from "../typechain-types";
+import { DAYS } from "./utils";
+
+describe("Aludel factory deployments", function () {
+  describe("WHEN deploying a template set", () => {
+    let factory: AludelFactory;
+    beforeEach(async () => {
+      await deployments.fixture(["templates"], {
+        keepExistingDeployments: true,
+      });
+      const deployedFactory = await deployments.get("AludelFactory");
+      factory = (await ethers.getContractAt(
+        deployedFactory.abi,
+        deployedFactory.address
+      )) as AludelFactory;
+    });
+    it("THEN GeyserV2 and AludelV1 reserved template addresses are added in a disabled state", async () => {
+      const geyserv2 = await factory.getTemplate(
+        "0x00000000000000000000000000000000be15efb2"
+      );
+      expect(geyserv2.disabled).to.be.true;
+      expect(geyserv2.name).to.be.equal("GeyserV2");
+      const aludelv1 = await factory.getTemplate(
+        "0x00000000000000000000000000000000a1fde1b1"
+      );
+      expect(aludelv1.disabled).to.be.true;
+      expect(aludelv1.name).to.be.equal("AludelV1");
+    });
+    it("AND an Aludel is added as an enabled template", async () => {
+      const aludelDeployment = await deployments.get("AludelV2");
+      const aludelTemplate = await factory.getTemplate(
+        aludelDeployment.address
+      );
+      expect(aludelTemplate.disabled).to.be.false;
+      expect(aludelTemplate.name).to.be.equal("AludelV2");
+    });
+
+    async function deployMockERC20(name: string): Promise<MockERC20> {
+      const factory = await ethers.getContractFactory("MockERC20");
+      return (await factory.deploy(name, "MockERC20")) as MockERC20;
+    }
+
+    describe("GIVEN a PowerSwitchFactory, a CrucibleFactory, a RewardPoolFactory, and some other stuff", () => {
+      let powerSwitchFactory: PowerSwitchFactory;
+      let crucibleFactory: CrucibleFactory;
+      let rewardPoolFactory: RewardPoolFactory;
+      let stakingToken: MockERC20;
+      let rewardToken: MockERC20;
+      beforeEach(async () => {
+        await deployments.fixture(
+          ["PowerSwitchFactory", "CrucibleFactory", "RewardPoolFactory"],
```

Comment:
I think we don't have a deployment for `CrucibleFactory`. We certainly need a Crucible Factory available to test these things but it's not clear from where the crucible factory comes from here.

Reply:
that's part of the magic of hardhat-deploy. You can either define a script for deploying a tag, or just add it as an already-existing contract (see `deployments/{hardhat,mumbai}/CrucibleFactory.json`) so `deployments.get(` can find it.

This means that hardhat will fail to do a deployment on a network without a CrucibleFactory deployment, which would be correct. This means we still require a forked node for running the JS tests, which isn't ideal, but also figuring out how to properly deploy the CrucibleFactory was outside of the scope of this PR and not yet strictly necessary